tofhir-engine/src/main/scala/io/tofhir/engine/model/FhirMappingResult.scala

```diff
@@ -5,44 +5,75 @@ import io.tofhir.engine.util.TimeUtil
 
 import java.sql.Timestamp
 
+/**
+ * Base trait representing the result of a FHIR mapping process.
+ */
+trait BaseFhirMappingResult {
+
+  /**
+   * Identifier of the job that this mapping is performed within
+   */
+  val jobId: String
+
+  /**
+   * The name of the specific mapping task being executed.
+   */
+  val mappingTaskName: String
+
+  /**
+   * Timestamp of the result
+   */
+  val timestamp: Timestamp
+
+  /**
+   * If there is a problem in the process, the JSON serialization of the source data
+   */
+  val source: String
+
+  /**
+   * If there is a problem in the process, description of the problem
+   */
+  val error: Option[FhirMappingError]
+
+  /**
+   * The optional identifier for FhirMappingJobExecution object
+   */
+  val executionId: Option[String]
+
+  /**
+   * The optional identifier for the project under which this mapping job is being executed.
+   */
+  val projectId: Option[String]
+}
 
 /**
- * Mapping process result
+ * Case class representing the result of a FHIR mapping process, extending the base mapping result.
```

Comment:
Can you please extend the description for this class?

FhirMappingResult class represents a single mappedFhirResource irrespective of how many FHIR resources is created from each row. Hence, different FhirMappingResult instances might be created from a single source row if the mapping is 1-to-many.